### PR TITLE
LC – BE-1: MoneyUtils Implementation with Tests

### PR DESCRIPTION
**Pull request by ShatterBloom**
Summary

- Added a unified MoneyUtils helper class under SettlyFinance.Utils
- Provides consistent money conversion (dollor ↔ cents) and rounding-to-cent rules
- Implements tail-adjustment logic to ensure the final installment clears remaining principal
- Added a dedicated test project SettlyFinanceTests with unit tests for MoneyUtils

Description of Changes
Implemented MoneyUtils with:

- ToCents(decimal) → Convert amount to cents (long), rounding to nearest cent
- FromCents(long) → Convert cents back to decimal amount
- RoundToCent(decimal) → Round any decimal amount to 2 decimal places
- FixPrincipalPart(...) → Adjust final installment principal to match remaining balance
- ReduceRemainingPrincipal(...) → Safely reduce remaining principal, preventing negative values

Created SettlyFinanceTests project to keep tests separated from API tests
Added unit tests covering:

- Conversion both ways (dollor ↔ cents)
- Edge rounding (e.g., 100.005 → 100.01)
- Tail adjustment for last vs non-last installments
- Principal clamping to zero

- - -
Perform an AI-assisted review on [CodePeer.com](https://codepeer.com/app/prs/github/SettlyAI-Group/SettlyAI/222)

## Summary by CodeRabbit

- **New Features**
  - Added a finance module providing precise money rounding, cent conversions, and installment principal helpers.

- **Tests**
  - Added comprehensive unit tests covering conversions, rounding, principal selection, progression, and edge cases.

- **Chores**
  - Integrated finance and test projects into the backend solution and configured test tooling.
  - Project structure updates: included migrations directory and removed the persisted EF Core model snapshot.

### CODE REVIEW

File: backend/SettlyFinance/Utils/MoneyUtils.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        //把金额四舍五入到 2 位小数（分）
+        public static decimal RoundToCent(decimal amount) =>
+            Math.Round(amount, 2, MidpointRounding.AwayFromZero);
+        //尾差修正：最后一期本金必须直接等于剩余本金，确保贷款归零
+        public static long FixPrincipalPart(long remainingPrincipalCents, long suggestedPrincipalPart, bool isLastInstallment)
+        {
+            if(!isLastInstallment)
+                return Math.Min(suggestedPrincipalPart, remainingPrincipalCents);
+            return remainingPrincipalCents;
+        }
```
[the remark this answers is not in the document]
Comment (ShatterBloom) [reply backend/SettlyFinance/Utils/MoneyUtils.cs]: Fixed as per suggestion.

File: backend/SettlyFinance/Utils/MoneyUtils.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        //更新剩余本金，保证不为负数
+        public static long ReduceRemainingPrincipal(long remainingPrincipalCents, long principalCents)
+        {
+            var next = remainingPrincipalCents - principalCents;
+            return next < 0 ? 0 : next;
+        }
```
[the remark this answers is not in the document]
Comment (ShatterBloom) [reply backend/SettlyFinance/Utils/MoneyUtils.cs]: Fixed as per suggestion.
